src/components/upload/FileDiv.tsx

```diff
@@ -59,12 +95,86 @@ interface Props {
   onRemoveFile: (file: string) => any;
 }
 
-const FileDiv: FC<Props> = ({ hash, fileName, onRemoveFile }: Props) => {
-  const [hover, setHover] = useState(false);
+const FileDiv: FC<Props> = ({
+  hash,
+  fileName,
+  fileBlob,
+  onRemoveFile,
+}: Props) => {
+  const [hover, setHover] = useState<boolean>(false);
+  const [keywords, setKeywords] = useState<string[]>([]);
+  const [numArticles, setNumArticles] = useState<number>(0);
+  const [numReferences, setNumReferences] = useState<number>(0);
+  const [loadingProgress, setLoadingProgress] = useState<number>(80);
+
+  const getKeywordsList = (text: string) => {
+    const identifier = "ID ";
+    const keywordsLines = text
+      .split("\n")
+      .filter((line) => line.startsWith(identifier));
+    return keywordsLines
+      .map((line) =>
+        line
+          .replace(identifier, "")
+          .trim()
+          .split(";")
+          .map((keyword) => keyword.trim().toLowerCase())
+          .filter((keyword) => Boolean(keyword))
+      )
+      .flat();
+  };
+
+  const mostCommonKeywords = (text: string, max: number = 3) => {
+    const keywordsList = getKeywordsList(text);
+    let count: { [keyword: string]: number } = {};
+    for (let keyword of keywordsList) {
+      count[keyword] = (count[keyword] ? count[keyword] : 0) + 1;
+    }
+    const sortCount = Object.entries(count).sort((first, second) =>
+      first[1] < second[1] ? 1 : -1
+    );
+    return sortCount.slice(0, max).map((item) => item[0]);
+  };
+
+  const countArticles = (text: string) => {
+    const identifier = "PT ";
+    return text.split("\n").filter((line) => line.startsWith(identifier))
+      .length;
+  };
+
+  const countReferences = (text: string) => {
+    const identifier = "NR ";
+    return text
+      .split("\n")
+      .filter((line) => line.startsWith(identifier))
+      .map((line) => parseInt(line.replace(identifier, "")))
+      .reduce((n, m) => n + m);
+  };
+
+  useEffect(() => {
+    fileBlob.text().then((text) => {
```

Comment:
Upload a file and analyze it on the front-end can be a really big task and can consume a lot of memory if the file is to big, so, I think that we can start using a limit of it size, and in the future we can move the logic of analyze the file to our backend, or what do you think?

Reply:
This processing is like nothing compared to the memory footprint of react, we might limit the number of files an user can upload tho.